Frogies-Project/Assets/Scripts/WaveSystem/WaveController.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Enemies;
+using StatsSystem.Health;
+using UnityEngine;
+using UnityEngine.Video;
+using Object = UnityEngine.Object;
+
+namespace WaveSystem
+{
+    public class WaveController
+    {
+        public event Action OnWaveCleared;
+        public event Action OnWaveStarted;
+        public event Action OnEnemyKilled;
+
+        private Dictionary<GameObject, bool> _enemyDictionary;
+        private List<GameObject> _spawners;
+        private List<GameObject> _enemies;
```

Comment:
the name of the variable is two generic, please rename to something like _enemiePerfabs, but I would rather put this data in the scriptable object